src/GooglePlacesApi/Services/GooglePlacesApiService.cs

```diff
@@ -40,15 +49,57 @@ public async Task<Predictions> GetPredictionsAsync(string searchText)
             return await _api.GetAutocompleteAsync(searchText, _settings.CreateQueryStringParameters())
                                  .ConfigureAwait(false);
         }
-
-
-        public async Task<Details> GetDetailsAsync(string placeId)
-        {
+
+        public async Task<Details> GetDetailsAsync(string placeId, string sessionToken, DetailLevel detailLevel = DetailLevel.Basic)
+        {
             if (string.IsNullOrWhiteSpace(placeId))

```

Comment:
Shouldn't there also be a check on sessionToken? Maybe if it is not null or empty? Or does it work even when sessionToken is empty?

Reply:
You also get results if the sessionToken is provided in the request like:
...[previous argument]&**sessiontoken=**&[next argument]...

Though I've implemented to always set a sessionToken in the service initialization:
```suggestion
   public GooglePlacesApiService(GoogleApiSettings settings) 
        {
        ...
           if (string.IsNullOrEmpty(settings.SessionToken))
                settings.SessionToken = Guid.NewGuid().ToString();
```
Therefore a sessionToken will always be provided.
There is a risk that you use this sessionToken too long, if you use the same initialized service for more then several minutes: in that case you may be billed individually per request by Google.